lib/puma/runner.rb

```diff
@@ -91,7 +91,10 @@ def close_control_listeners
     end
 
     # @!attribute [r] ruby_engine
+    # @deprecated Use `RUBY_DESCRIPTION` instead
     def ruby_engine
+      warn "Puma::Runner#ruby_engine is deprecated; use RUBY_DESCRIPTION instead. It will be removed in puma v7."
```

Comment:
```
$ ruby -I lib -rpuma -rpuma/runner -e 'p Puma::Runner.new(Puma::Launcher.new(Puma::Configuration.new)).ruby_engine'
Puma::Runner#ruby_engine is deprecated; use RUBY_DESCRIPTION instead. It will be removed in puma v7.
"ruby 3.3.2-p78"
```